spec.html

```diff
@@ -32118,11 +32204,25 @@ <h1>
         </h1>
         <dl class="header">
           <dt>description</dt>
-          <dd>It converts _t_ from UTC to local time.</dd>
+          <dd>
+            It converts _t_ from UTC to local time.
+            The local political rules for standard time and daylight saving time in effect at _t_ should be used to determine the result in the way specified in this section.
+          </dd>
         </dl>
         <emu-alg>
-          1. Return _t_ + LocalTZA(_t_, *true*).
+          1. Let _localTimeZone_ be DefaultTimeZone().
+          1. Let _offsetParseResult_ be Completion(ParseTimeZoneOffsetString(_localTimeZone_)).
+          1. If _offsetParseResult_ is a normal completion, then
+            1. Let _offsetNs_ be _offsetParseResult_.[[Value]].
+          1. Else,
+            1. Let _offsetNs_ be GetNamedTimeZoneOffsetNanoseconds(_localTimeZone_, ℤ(ℝ(_t_) &times; 10<sup>6</sup>)).
```

Comment:
I think we need an assertion that `_t_` is not `NaN` so that we can safely take `ℝ(_t_)`.

Reply:
That should be fine, I just checked that all callers of LocalTime don't call it with NaN.

Reply:
I'd also consider adding "If _t_ is **NaN**, return **NaN**" to UTC(), but that has pre-existing normative implications, so maybe should be out of scope for this PR.

(Specifically, `new Date(NaN, NaN)` ends up calling UTC(NaN) which computes NaN - LocalTZA(NaN, false), and LocalTZA did not specify what to do with NaN. Web reality seems to be that UTC(NaN) should return NaN.)

Reply:
You said "That should be fine", but you didn't actually add the assertion. In addition to excluding NaN, you presumably need to exclude infinities, so I think you should change the type of `_t_` from `a Number` to `a finite Number`.

Reply:
For LocalTime(), I changed the type in the structured header to "a finite time value", which is implicitly an assertion.

For UTC(), see my comment after the "that should be fine" one — I don't think it's possible to add an assertion to UTC() here without contradicting web reality, and I would like to place resolving that pre-existing discrepancy out of scope of this PR.

Reply:
I understand your desire to leave something to another PR, but do you think that the proposed definition of UTC() *doesn't* contradict web reality?

Reply:
I think there's an error in the spec either way: with the status quo and this PR, the error is trying to do arithmetic wiht a NaN; with a tighter type for `_t_`, the error is trying to pass a NaN to an AO that doesn't accept it.

I'm fine with whatever the editors decide.